### PR TITLE
support for sendmail

### PR DESCRIPTION
**Pull request by Sebseb01**
### Hello, i have used your role, but with sendmail. i send to you my modification, i dont know if your interested. (sorry if some basic error i'm a pretty new user of ansible) thanks in advance.

### CODE REVIEW

**Comment by foxycode** [comment]
@Sebseb01 Those sendmail refresh command looks pretty foreign to me. Are you using Ubuntu? This role was meant for Debian/Ubuntu only. Is this common way to refresh sendmail? Is make installed automaticcally when you have sendmail on your system? I believe it isn't.

**Comment by Sebseb01** [comment]
It's beacause sendmail use m4 macro for generate sendmail.cf 


m4 and make are in dependencies of package sendmail (and specialy on sendmail-cf)

I have use your role and testmy pull-request on Debian stretch and Buster,  i don't have any reason to think it will  not the same on ubuntu. I wiil be test later and commit a fix if it doesn't work.

**Comment by foxycode** [comment]
Let me know when you test it then ;) Also you should add sendmail `env` to `.travis.yml` for automated testing.

**Comment by ulvida** [comment]
Heureux de savoir que ce travail a servi, et plus encore de constater que l'on travaille sur des technologies accessibles! 

Thanks for your contribution! I agree with @foxycode to test well before merging, but think it's good to integrate other MTAs' management in the role. Maybe we can help to test your PR in Ubuntu.

**Comment by Sebseb01** [comment]
I have just tested with Xenial and it perfectly works

**Comment by Sebseb01** [comment]
I have tried to use travis without success for the moment.

Je contribue avec plaisire, votre "code" d'une grande qualité, m'apprend quantités de choses.
Le texte brute est très accessible, jai la chance d'avoir choisi ce métier (avant le handicap), qui est aussi une passion.

;-)

**Comment by foxycode** [comment]
Guys, I tried to test with Debian before with o success. It's different to run role on system than run role on system you want to setup. I had problem to add Debian to test targets because of default Ansible version,  If you figure it out, it would be great!

**Comment by Sebseb01** [comment]
Ok, it was just a typping error. Please let me know if you have some issue.

For travis on debian buster you can remove the ppa ... in debian repo it's ansible 2.7 and i think your role work fine with this version. or activate backport for 2.9. and for stretch you have 2.7 in backports.

**Comment by foxycode** [comment]
@Sebseb01 I don't use Debian, so it would great if you could fix tests for Debian. Looks like you have experience.

Also, add sendmail test for all Ubuntu version in `.travis.yml` please.

**Comment by Sebseb01** [comment]
Ok @foxycod, I have added and passed of all ubuntu version and debian too.

I hope this can help you.

File: .travis.yml [hunk 1]
```diff
@@ -28,6 +28,21 @@ env:
...
+    playbook: sendmail.yml
+  - distribution: ubuntu
+    version: focal
+    playbook: sendmail.yml
+  - distribution: debian
+    version: stretch
+    playbook: sendmail.yml
+  - distribution: debian
+    version: buster
+    playbook: sendmail.yml
```
Comment (foxycode) [inline .travis.yml:58]: Test sendmail with Ubuntu too please:
```
  - distribution: ubuntu
    version: xenial
    playbook: same_key.yml
  - distribution: ubuntu
    version: xenial
    playbook: different_keys.yml
  - distribution: ubuntu
    version: xenial
    playbook: sendmail.yml
  - distribution: ubuntu
    version: bionic
    playbook: same_key.yml
  - distribution: ubuntu
    version: bionic
    playbook: different_keys.yml
  - distribution: ubuntu
    version: bionic
    playbook: sendmail.yml
  - distribution: ubuntu
    version: focal
    playbook: same_key.yml
  - distribution: ubuntu
    version: focal
    playbook: different_keys.yml
  - distribution: ubuntu
    version: focal
    playbook: sendmail.yml
  - distribution: debian
    version: jessie
    playbook: same_key.yml
  - distribution: debian
    version: jessie
    playbook: different_keys.yml
  - distribution: debian
    version: jessie
    playbook: sendmail.yml
  - distribution: debian
    version: stretch
    playbook: same_key.yml
  - distribution: debian
    version: stretch
    playbook: different_keys.yml
  - distribution: debian
    version: stretch
    playbook: sendmail.yml
  - distribution: debian
    version: buster
    playbook: same_key.yml
  - distribution: debian
    version: buster
    playbook: different_keys.yml
  - distribution: debian
    version: buster
    playbook: sendmail.yml
```

**Review by foxycode — CHANGES_REQUESTED** [review]
You did a great work, thank you! Before I merge it, need to to fix few minor details I hope.

* Update test env as I suggested please, to test `sendmail` with `Ubuntu` too.
* I added `jessie` too, because it should be supported. If you can't make it work on `jessie`, remove `jessie` from supported versions in `meta/main.yml` please.

Otherwise it looks very good :+1:

**Comment by Sebseb01** [comment]
Ok i have edit the travis.yml 

I think it's a good idea to remove support of  Jessie this version is no more supported by the LTS project and it very old now.

File: meta/main.yml [hunk 2]
```diff
@@ -15,13 +15,13 @@ galaxy_info:
...
     versions:
-    - jessie
     - stretch
     - buster
   galaxy_tags:
     - mail
     - dkim
     - antispam
-    - postfix
+    - postfix    - postfix
```
Comment (foxycode) [inline meta/main.yml]: This is probably typo

**Comment by foxycode** [comment]
Can you please look why `xenial` tests failed?

**Comment by Sebseb01** [comment]
Hello, I have created a Xenial  container in my local test, and manually running test. It just work. I don't understand why this failure apear after i have edit "meta".

And why systemd can't start dkim service ... 

i don't have idea for the moment

**Comment by foxycode** [comment]
Ok, since it worked for you, I will merge it anyway.

Many thanks for your contributtion1

**Pull request merged**